src/main/java/edu/harvard/iq/dataverse/engine/command/impl/ReconcileDatasetPidCommand.java

```diff
@@ -0,0 +1,181 @@
+                    Collections.singleton(Permission.EditDataset), getDataset());
+        }
+        // Datast must be unreleased! This means there is only one version!
+        if (getDataset().isReleased()) { //@TODO: Clarify whether this is the best check...
+            throw new IllegalCommandException("Dataset already published, cannot alter PID Provider", this);
+        }
+        // Dataset must not be harvested!
+        if (getDataset().isHarvested()) {
+            throw new IllegalCommandException("Dataset is harvested, cannot alter PID Provider", this);
+        }
+        GlobalId oldId = getDataset().getGlobalId();
+
+        if (oldId == null) {
+            throw new IllegalStateException("Dataset without a global identifier, cannot alter!");
+        }
+        PidProvider currentPidProvider = PidUtil.getPidProvider(oldId.getProviderId());
+        // new PID Provider must be different to requested one!
+        if (this.newPidProvider.equals(currentPidProvider)) {
+            throw new IllegalCommandException("PID Provider " + currentPidProvider.getId() + " is same as configured. This Operation has no effect!", this);
+        }
+
+        logger.fine("Reconciling dataset( id =`" + getDataset().getId() + ")` - removing globalId `" + getDataset().getGlobalId() + '`');
+        // remove dataset PID
+        try {
+            if (currentPidProvider.alreadyRegistered(getDataset())) { //if not registered with PIDProvider than there is no need to delete it...
+                currentPidProvider.deleteIdentifier(getDataset()); // delete it externally
+            }
+            getDataset().setGlobalId(null); //  remove it internally
+            getDataset().setGlobalIdCreateTime(null);
+            getDataset().setIdentifierRegistered(false);
+        } catch (Exception e) {
+            logger.log(Level.WARNING, "Identifier deletion was not successful:", e.getMessage());
+        }
+
+        if(ctxt.systemConfig().isFilePIDsEnabledForCollection(getDataset().getOwner())) {
+            reconcileFilePids(ctxt);
+        }
+
+        newPidProvider.generatePid(getDataset()); // this updates Protocol, Authority, and Identifier and thus a new GlobalID
+        logger.fine("Reconciling dataset( id =`" + getDataset().getId() + ")` - creating new globalId `" + getDataset().getGlobalId() + '`');
+        if (!newPidProvider.registerWhenPublished()) {
+            registerExternalIdentifier(getDataset(), ctxt, true); // this updates GlobalIdCreateTime and IdentifierRegistered
+        }
+        // keep old Id als alternative identifier
+        AlternativePersistentIdentifier api;
+        api = new AlternativePersistentIdentifier();
+        api.setProtocol(oldId.getProtocol());
+        api.setAuthority(oldId.getAuthority());
+        api.setIdentifier(oldId.getIdentifier());
+        api.setDvObject(getDataset());
+        api.setStorageLocationDesignator(true);// cf. Dataset#getIdentifierForFileStorage()
```

Comment:
rare case - if there's already an alt PID that is the storageLocationDesignator

Reply:
I've added needed code and added a testcase.